src/js/components/returns/inbound/AddItemsPage.jsx

```diff
@@ -412,25 +462,16 @@ class AddItemsPage extends Component {
       });
   }
 
-  previousPage(values, invalid) {
+  async previousPage(values, invalid) {
     if (!invalid) {
-      this.saveStockTransferInCurrentStep(values.returnItems)
-        .then(() => this.props.previousPage(this.state.inboundReturn));
+      await this.saveStockTransfer(values.returnItems, null);
     } else {
-      confirmAlert({
-        title: this.props.translate('react.inboundReturns.confirmPreviousPage.label', 'Validation error'),
-        message: this.props.translate('react.inboundReturns.confirmPreviousPage.message', 'Cannot save due to validation error on page'),
-        buttons: [
-          {
-            label: this.props.translate('react.inboundReturns.confirmPreviousPage.correctError.label', 'Correct error'),
-          },
-          {
-            label: this.props.translate('react.inboundReturns.confirmPreviousPage.continue.label', 'Continue (lose unsaved work)'),
-            onClick: () => this.props.previousPage(this.state.inboundReturn),
-          },
-        ],
-      });
+      const correctErrors = await this.confirmValidationErrorOnPreviousPage();
+      if (correctErrors) {
+        return;
+      }
```

Comment:
maybe `return` without else will be more readable?
\+ with that change `if (correctErrors)` won't be nested in the else.